### ISSUE TITLE
Parallel Route Modal closed using router.push cannot open again until hard refresh

### ISSUE THREAD

**Issue by martitv** [vercel/next.js#66025]
### Link to the code that reproduces this issue

https://github.com/martitv/ModalPushTest

### To Reproduce

1. Start the application with `pnpm dev`
2. Click a photo.
3. Click "Go back!" button in modal

### Current vs. Expected behavior

Current:
Modal doesn't close.
In my real world app, the modal closes. But you cannot open it again until hard refresh. It just changes the route, but nothing happens. Cant replicate as in this test app the modal fails earlier.

Expected:
Modal closes.
In my real world app I would expect to be able to open a modal again when clicking a link matching the paralell route

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Enterprise
  Available memory (MB): 16164
  Available CPU cores: 8
Binaries:
  Node: 18.17.1
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: 14.1.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.2
Next.js Config:
  output: standalone
```

### Which area(s) are affected? (Select all that apply)

Parallel & Intercepting Routes

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by martitv** [comment]
I believe the issue might be connected to portals? Got the inspiration from this example:
https://ariakit.org/examples/dialog-next-router

I tried removing the portal target in my real-world app and now it works. Using mantine modal in that app. But i also tried using the nextGram modal which gave the same behaviour when removing the portal.

**Comment by Sam-Apostel** [comment]
I seem to be having the same problem using `redirect` from a server action. The react component does not get remounted after redirection ('push' / 'replace') to close it.

`router.back('/')` does still work as expected

**Comment by rthaut** [comment]
I just ran into this same issue (currently using `next@14.2.5`) following the ["Modals" Example for Parallel Routes"](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes#closing-the-modal).

Using `router.back()` closes the modal, but `router.push('/')` and `router.replace('/')` both fail to close the modal, despite updating the URL as expected.

I have both a `default.tsx` and a `[...catchAll]/page.tsx` that both return `null` in the @​slot.

For now I am using `window.location.href = '/'` to force the hard-refresh, thereby ensuring the modal closes and allowing me to specify the target path (which is why I can't rely on `router.back()` in my use case).

I can work up a minimum code repository this week, unless someone already has a working example to provide.

**Comment by 50Bytes-dev** [comment]
Same problem

**Comment by pablojsx** [comment]
man... parallel routes and route intercept are very problematic. I have 2 similar issues: I need to bypass the intercepted route and navigate directly to the page but it's not possible without using <a>. And another issue is similar to what the OP mentioned, parallel routes depends on router.back..I need to navigate to a different URL but the modal won't open again, I guess because I didn't use router.back?it keeps track of the open state by going back

**Comment by edproton** [comment]
The errors keeps...I see issues related with since February.

**Comment by edproton** [comment]
> I just ran into this same issue (currently using `next@14.2.5`) following the ["Modals" Example for Parallel Routes"](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes#closing-the-modal).
> 
> Using `router.back()` closes the modal, but `router.push('/')` and `router.replace('/')` both fail to close the modal, despite updating the URL as expected.
> 
> I have both a `default.tsx` and a `[...catchAll]/page.tsx` that both return `null` in the @​slot.
> 
> For now I am using `window.location.href = '/'` to force the hard-refresh, thereby ensuring the modal closes and allowing me to specify the target path (which is why I can't rely on `router.back()` in my use case).
> 
> I can work up a minimum code repository this week, unless someone already has a working example to provide.

Ah, the infamous parallel routes bug in Next.js on Vercel strikes again! No wonder the community jokes that "stability" in Next.js is just a polite suggestion. I just spent two hours on a wild goose chase, wrestling with `catchAll`, only to end up in `default.tsx`—like I’ve been exiled to Next.js limbo. And the cherry on top? `router.back()` is a team player, while `router.push('/')` is off sipping lattes, refusing to work. Classic. 😅  

Oh, and here’s a similar issue that’s been marinating for a year:  
https://github.com/vercel/next.js/issues/62430

**Comment by CombeeMike** [comment]
Same issue here with the difference that my modal is closing correctly but it also won't re-open when navigating to the route again...

I'm at Next `v14.2.17` and can't update to `v15` ATM b/c of some dependencies to React 18. Does anyone know if I can hope for this to be not an issue anymore when updating to `v15`?

**Comment by ztanner** [comment]
Hi folks,

In the provided example, a simple fix would be to add a `app/@modal/page.tsx` that returns `null`.

A catch-all route (for ex, `[...catchAll]`) will match all routes except for the "index" route that it's defined in. Meaning /foo, /bar, /foo/bar, ...etc, but not /.

With parallel routes, during a soft navigation (ie when clicking a Link, or `router.back()`) the router will attempt to match the page to a corresponding parallel route. Since there's no matching page component for the `@modal` parallel route on the `/` path, it's keeping whatever was previously in that slot, until you refresh the page. You can learn about that behavior [here](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes#active-state-and-navigation).
